config/config_test.go

```diff
@@ -427,3 +427,53 @@ func (s *ConfigTestSuite) TestConfigPeriod() {
 		s.Equal(test.expectedPeriod, c.Period)
 	}
 }
+
+func (s *ConfigTestSuite) TestActivatedRulesFrameWork() {
+	var tests = []struct {
+		config                 string
+		expectedActivatedRules []string
+		expectedType           string
+	}{
+		{
+			`
+type: cloudbeat/vanilla
+streams:
+  - data_yaml:
+      activated_rules:
+        cis_k8s:
+          - a
+          - b
+`,
+			[]string{"a", "b"},
+			"cloudbeat/vanilla",
+		},
+		{
+			`
+type: cloudbeat/eks
+streams:
+  - data_yaml:
+      activated_rules:
+        cis_eks:
+          - a
+          - b
+`,
+			[]string{"a", "b"},
+			"cloudbeat/eks",
+		},
+	}
+
+	for _, test := range tests {
+		cfg, err := config.NewConfigFrom(test.config)
+		s.NoError(err)
+
+		c, err := New(cfg)
+		s.NoError(err)
+
+		s.Equal(test.expectedType, c.Type)
+		if test.expectedType == "cloudbeat/eks" {
```

Comment:
Can the test checks both benchmarks at the same time?

For example:
Instead of having `expectedActivatedRules []string`, we can have `expectedEKSActivatedRules []string` and `expectedVanillaActivatedRules []string`

And then in the assert, we will assert them both.

That way I think that test will "lock" the behavior even better.

WDT?

Reply:
asserting them both can result in false positive in the case where both of the configuration resulted only EKS or Vanilla

Reply:
```suggestion
		func (s *ConfigTestSuite) TestActivatedRulesFrameWork() {
	var tests = []struct {
		config                    string
		expectedActivatedRules    []string
		expectedEksActivatedRules []string
		expectedType              string
	}{
		{
			`
type: cloudbeat/vanilla
streams:
  - data_yaml:
      activated_rules:
        cis_k8s:
          - a
          - b
`,
			[]string{"a", "b"},
			nil,
			"cloudbeat/vanilla",
		},
		{
			`
type: cloudbeat/eks
streams:
  - data_yaml:
      activated_rules:
        cis_eks:
          - a
          - b
`,
			nil,
			[]string{"a", "b"},
			"cloudbeat/eks",
		},
	}

	for _, test := range tests {
		cfg, err := config.NewConfigFrom(test.config)
		s.NoError(err)

		c, err := New(cfg)
		s.NoError(err)

		s.Equal(test.expectedType, c.Type)
		s.Equal(test.expectedActivatedRules, c.Streams[0].DataYaml.ActivatedRules.CisK8s)
		s.Equal(test.expectedEksActivatedRules, c.Streams[0].DataYaml.ActivatedRules.CisEKS)
	}
}
```

Reply:
What do you think about this option?

Reply:
it's the same, I see no point in changing it, just leave it

Reply:
I don't think that they are the same option, since the first option doesn't check that the other configuration is actually `Nil` and not an empty array or anything else.

It will just "lock" the behavior

Reply:
oh this is what you mean by lock ok